apps/picsa-apps/dashboard/src/app/modules/translations/translations.service.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable } from '@angular/core';
+// eslint-disable-next-line @nx/enforce-module-boundaries
+import { Database } from '@picsa/server-types';
+import { PicsaAsyncService } from '@picsa/shared/services/asyncService.service';
+// import { PicsaNotificationService } from '@picsa/shared/services/core/notification.service';
+import { SupabaseService } from '@picsa/shared/services/core/supabase';
+import { IStorageEntry } from '@picsa/shared/services/core/supabase/services/supabase-storage.service';
+
+export type ITranslationRow = Database['public']['Tables']['translations']['Row'];
+
+export interface IResourceStorageEntry extends IStorageEntry {
+  /** Url generated when upload to public bucket (will always be populated, even if bucket not public) */
+  publicUrl: string;
+}
+@Injectable({ providedIn: 'root' })
+export class TranslationDashboardService extends PicsaAsyncService {
+
+  public  translations: ITranslationRow[] = [];
+
+  public get table() {
+    return this.supabaseService.db.table('translations');
+  }
+
+  constructor(private supabaseService: SupabaseService) {
+    super();
+  }
+
+  public override async init() {
+    await this.supabaseService.ready();
+    await this.listTranslations();
+  }
+
+
+  public async listTranslations() {
+    const { data, error } = await this.supabaseService.db.table('translations').select<'*', ITranslationRow>('*');
+    if (error) {
+      throw error;
+    }
+    this.translations = data || [];
+  }
+   
+  // update a translation record by ID
+  public async updateTranslationById(id: number, updatedData: Partial<ITranslationRow>): Promise<string> {
+    const { data, error } = await this.supabaseService.db.table('translations').update(updatedData).eq('id', id);
```

Comment:
nit(non-blocking) - up till now the way ids have been generated for app translations is a combination of the text in english along with 2 metadata columnns titled `tool` and `context`. This means that we avoid duplicate translations if something like the word "close" appears multiple times (e.g. lots of tools have dialogs with a close button), but allows for multiple translations where required (e.g. "close" dialog vs the distance is not far but is "close").

So what this means in practical terms is instead of searching for an id we will need to search for a composite match of en text with context and tool data fields. 

Fortunately supabase supports this via chaining query syntax, e.g. `.eq('en', en).eq('tool',tool).eq('context',context)`

However I don't think this is need right now, as I think possible we will want to refactor to remove the edit page and allow users to directly edit on the main page (possibly from the table itself) - so suggest no further action required right now but just wanted to share the information. Hope that makes sense

Reply:
Thanks for this.